pkg/util/fast_int_set.go

```diff
@@ -17,10 +17,11 @@ package util
 import (
 	"bytes"
 	"fmt"
+	"sort"
 )
 
-// FastIntSet keeps track of a set of integers. It is very fast when the values
-// are small. It is not thread-safe.
+// FastIntSet keeps track of a set of integers. It does not perform any
+// allocations when the values are small. It is not thread-safe.
 type FastIntSet struct {
 	smallVals uint64
 	largeVals map[uint32]struct{}
```

Comment:
All right, but you could still have a map[int32]int32  and pack multiple bits per map entry instead of map[int64]struct{}.